### PR TITLE
Improvments 2.0.0

### PR DESCRIPTION
**Pull request by danielgrigore**
Hi,
This is a pull request from 1and1/Calendar team.
The changes we did can simply deducted from the commit messages but I will emphasize them better here:

- In the TimeZoneRegistryImpl class we added the possibility to generate a VTimezone object when there isn't one already. To be able to this we took advantage of the new "java.time" package that was added in Java 8. For this reason we also changed the sourceCompatibility in build.gradle from 1.6 to 1.8.
- Because many calendars we parse from external sources do not strictly comply with RFC 5545 specs we added the possibility to fix some of the broken fields before validating the Calendar object. This is extendable and optional and is done in Calendar class invoking method "conformToRfc5545"
- Added ACKNOWLEDGED property.
- Because we serialize/deserialize calendar objects and the transient fields are no longer initialized we fixed this by adding the standard method private void readObject(ObjectInputStream in) in ComponentFactoryImpl, ParameterFactoryImpl, PropertyFactoryImpl

If any questions do not hesitate to contact me.
Cheers,

### CODE REVIEW

**Comment by benfortuna** [comment]
Thanks Daniel, I can see some good improvements and it's most appreciated.

One thing that might be an issue is the change to minimum of java 8. I think there's still quite a few people on Java 7 at least, so I might see if there's a way to support earlier jvms. If the change requires the new date/time API I think there is a backport dependency (jsr-310) that should provide that functionality.

**Comment by danielgrigore** [comment]
Hi Ben,
Thank you for the reply.

As I said in my first comment Java 8 was needed to generate a VTimezone when one could not be found. This is necessary when generating ics files that are sent as invitations to Outlook users. Outlook mail client does not correctly display an invitation with Accept/Tentative/Deny buttons if a correct timezone is not found in the ics file.

I think that moving from Java 7 to Java 8 is a good thing due the fact that Oracle stopped supporting Java 7  public updates in 2015 and this means that people using it without buying support,  won't receive any kind of updates, including security updates.
See https://java.com/en/download/faq/java_7.xml

Cheers,

File: src/main/java/net/fortuna/ical4j/model/Rfc5545Rule.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Incarnation of a RFC5545 rule.
+ * 
+ * @author daniel grigore
+ * @author corneliu dobrota
+ *
+ * @param <T>
+ *            type of the element this rule can be applied to
+ */
+public interface Rfc5545Rule<T> {
```
Comment (benfortuna) [inline src/main/java/net/fortuna/ical4j/model/Rfc5545Rule.java]: @danielgrigore I think these transformation rules would be better suited under the net.fortuna.ical4j.transform package. This package has long been neglected but was intended for applying mutations to calendar/component/property objects.

File: src/main/java/net/fortuna/ical4j/model/property/Aknowledged.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+Daboo                   Expires December 11, 2012              [Page 11]
+ 
+Internet-Draft              VALARM Extensions                  June 2012
+
+
+   Example:  The following is an example of this property:
+
+   ACKNOWLEDGED:20090604T084500Z
+ */
+public class Aknowledged extends UtcProperty{
```
Comment (benfortuna) [inline src/main/java/net/fortuna/ical4j/model/property/Aknowledged.java]: Typo in the class name

File: src/main/resources/zoneinfo/Australia/Sydney.ics [hunk 3]
```diff
@@ -3,128 +3,21 @@ PRODID:-//tzurl.org//NONSGML Olson 2016j//EN
 VERSION:2.0
 BEGIN:VTIMEZONE
 TZID:Australia/Sydney
-TZURL:http://tzurl.org/zoneinfo/Australia/Sydney
+TZURL:http://tzurl.org/zoneinfo-outlook/Australia/Sydney
```
Comment (benfortuna) [inline src/main/resources/zoneinfo/Australia/Sydney.ics]: We have a separate library for the zoneinfo-outlook definitions.

File: build.gradle [hunk 4]
```diff
@@ -22,8 +22,8 @@ apply plugin: 'signing'
 apply plugin: 'pl.allegro.tech.build.axion-release'
 apply from: 'gradle/dist.gradle'
 
-sourceCompatibility = 1.7
-targetCompatibility = 1.7
+sourceCompatibility = 1.8
```
Comment (benfortuna) [inline build.gradle]: I think we should be able to use the jsr-310 backport library without explicitly required java 8 as minimum

**Comment by danielgrigore** [comment]
Hi Ben,
I recently merged your PR-requests (the ones related to java8 time package and transform package) and included them in this pull request.
Currently the project builds fine with java 7 and tests pass.
Please let me know if there's something else I could help you with.

Cheers,

**Comment by benfortuna** [comment]
@danielgrigore thanks, I'll review and merge the changes shortly. Many thanks for the contribution.

**Pull request merged**